data_quality/generators/base.py

```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+
+class BaseGenerator(object):
+    """This is the base class for generators. All generators should inherit."""
+
+    def __init__(self, url=None):
+
+        self.base_url = url
+        if not self.base_url:
+            raise TypeError('Cannot generate the database without the "url" parameter.')
+
+    def generate_sources(self, sources_filepath, file_types=['csv','excel']):
```

Comment:
If you are creating a base class like this in order to define the interface for a generator, you should at least raise `NotImplementedError` for each required method here on the base class.